snapcraft/internal/pluginhandler.py

```diff
@@ -190,7 +190,7 @@ def is_dirty(self, step):
             # YAML (taken from self.code.options). If they've changed, then
             # this step is dirty and needs to run again.
             if state.properties != state.properties_of_interest(
-                    self.code.options):
+                    self.code.options.__dict__):
```

Comment:
Even though this code is transitional I will do it ;-)

I personally prefer __dict__